### ISSUE TITLE
Unable to unmap default `zg`

### ISSUE THREAD

**Issue by simvux** [neovim/neovim#21362]
### Neovim version (nvim -v)

v0.8.1

### Vim (not Nvim) behaves the same?

yes, vim 9.0.948

### Operating system/version

Void Linux

### Terminal name/version

Alacritty

### $TERM environment variable

alacritty

### Installation

System package manager

### How to reproduce the issue

`unmap zg`

### Expected behavior

[The default binding](https://neovim.io/doc/user/spell.html#zg) for `zg` should be unmapped, so that it's possible to bind it to a different action. 

### Actual behavior

even though `zg` works in the default configuration, it doesn't seem to appear in the list provided by `:map`. 

attempting to unmap it gives `E31: No such mapping`

attempting to create a new map with `zg` has no effect since the default behavior is still active. 

Is it some special kind of mapping that cannot be overwritten?

**Comment by lewis6991** [comment]
> [The default binding](https://neovim.io/doc/user/spell.html#zg) for zg should be unmapped, so that it's possible to bind it to a different action.

Why should it be unmapped? It is a built-in binding. To unmap it you need to bind it to `<nop>`

**Comment by simvux** [comment]
my goal was to use the `z` key as the modifier for LSP actions. 

so
`zg` == `goto definition`
`zd` == `diagnostic list`
`zr` == `rename`
etc. 

However; I just discovered that just creating a new mapping *does* work and the problem was actually that the `on_attach` function I used for configuration LSP bindings only when a language server is active didn't actually run. So some of the bindings didn't work.
